src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/BinaryFormat/Deserializer/Deserializer.cs

```diff
@@ -75,34 +75,34 @@ internal sealed partial class Deserializer : IDeserializer
     // dictionary, to do so we'd need to know if any referenced object is going to get to this state
     // even if it hasn't finished parsing, which isn't easy to do with cycles involved.
     private Queue<PendingSerializationInfo>? _pendingSerializationInfo;
-    private HashSet<int>? _pendingSerializationInfoIds;
+    private HashSet<SerializationRecordId>? _pendingSerializationInfoIds;
 
     // Keeping a separate stack for ids for fast infinite loop checks.
-    private readonly Stack<int> _parseStack = [];
+    private readonly Stack<SerializationRecordId> _parseStack = [];
```

Comment:
This will significantly hurt performance for deep graphs if `SerializationRecordId` is not `RuntimeHelpers.IsBitwiseEquatable<T>()`. If it isn't, or can't be made so, we should allow getting the raw value.

Reply:
PTAL at https://github.com/dotnet/runtime/pull/103232/commits/daa743115d02e9142101f41985ec4f808fbdb6a6, I think I was able to remove the need for having this separate stack

Reply:
I think you can just throw there.